SemiconductorTestLibrary.Extensions/source/InstrumentAbstraction/DCPower/Source.cs

```diff
@@ -222,7 +298,121 @@ public static void ForceCurrent(this DCPowerSessionsBundle sessionsBundle, doubl
+        /// </summary>
+        /// <param name="sessionsBundle">The <see cref="DCPowerSessionsBundle"/> object.</param>
+        /// <param name="currentLevels">The current level to force for different site-pin pairs.</param>
+        /// <param name="voltageLimit">The voltage limit to use.</param>
+        /// <param name="currentLevelRange">The current level range to use.</param>
+        /// <param name="voltageLimitRange">The voltage limit range to use.</param>
+        /// <param name="waitForSourceCompletion">Setting this to True will wait until sourcing is complete before continuing, which includes the set amount of source delay.
+        /// Otherwise, the source delay amount is not directly accounted for by this method and the WaitForEvent must be manually invoked in proceeding code.</param>
+        public static void ForceCurrent(this DCPowerSessionsBundle sessionsBundle, PinSiteData<double> currentLevels, double? voltageLimit = null, double? currentLevelRange = null, double? voltageLimitRange = null, bool waitForSourceCompletion = false)
+        {
+            sessionsBundle.Do((sessionInfo, sitePinInfo) =>
+            {
+                var settings = new DCPowerSourceSettings()
+                {
+                    OutputFunction = DCPowerSourceOutputFunction.DCCurrent,
+                    LimitSymmetry = DCPowerComplianceLimitSymmetry.Symmetric,
+                    Level = currentLevels.GetValue(sitePinInfo.SiteNumber, sitePinInfo.PinName),
+                    Limit = voltageLimit,
+                    LevelRange = currentLevelRange,
+                    LimitRange = voltageLimitRange
+                };
+                sessionInfo.Force(settings, sitePinInfo.IndividualChannelString, waitForSourceCompletion: waitForSourceCompletion);
+            });
+        }
+
+        /// <summary>
+        /// Forces current and specifies voltage limit.
+        /// </summary>
+        /// <param name="sessionsBundle">The <see cref="DCPowerSessionsBundle"/> object.</param>
+        /// <param name="settings">The settings to use.</param>
+        /// <param name="waitForSourceCompletion">Setting this to True will wait until sourcing is complete before continuing, which includes the set amount of source delay.
+        /// Otherwise, the source delay amount is not directly accounted for by this method and the WaitForEvent must be manually invoked in proceeding code.</param>
+        public static void ForceCurrent(this DCPowerSessionsBundle sessionsBundle, DCPowerSourceSettings settings, bool waitForSourceCompletion = false)
+        {
+            sessionsBundle.Do(sessionInfo =>
+            {
+                settings.OutputFunction = DCPowerSourceOutputFunction.DCCurrent;
+                sessionInfo.Force(settings, sessionInfo.AllChannelsString, waitForSourceCompletion);
+            });
+        }
+
+        /// <summary>
+        /// Forces current and specifies voltage limit.
```

Comment:
Suggest changing to: "Forces current using specified site-unique source settings."